crates/core/src/db/datastore/locking_tx_datastore/tx.rs

```diff
@@ -30,38 +31,37 @@ impl StateView for TxId {
         self.committed_state_shared_lock.table_row_count(table_id)
     }
 
-    fn iter<'a>(&'a self, ctx: &'a ExecutionContext, table_id: TableId) -> Result<Iter<'a>> {
-        self.committed_state_shared_lock.iter(ctx, table_id)
+    fn iter(&self, table_id: TableId) -> Result<Iter<'_>> {
+        self.committed_state_shared_lock.iter(table_id)
     }
 
     /// Returns an iterator,
     /// yielding every row in the table identified by `table_id`,
     /// where the values of `cols` are contained in `range`.
-    fn iter_by_col_range<'a, R: RangeBounds<AlgebraicValue>>(
-        &'a self,
-        ctx: &'a ExecutionContext,
+    fn iter_by_col_range<R: RangeBounds<AlgebraicValue>>(
+        &self,
         table_id: TableId,
         cols: ColList,
         range: R,
-    ) -> Result<IterByColRange<'a, R>> {
+    ) -> Result<IterByColRange<'_, R>> {
         match self.committed_state_shared_lock.index_seek(table_id, &cols, &range) {
             Some(committed_rows) => Ok(IterByColRange::CommittedIndex(CommittedIndexIter::new(
-                ctx,
+                &self.ctx,
```

Comment:
Just pointing out that this is dead code too.

Reply:
Done